### ISSUE TITLE
diamond killed due to exceeding swap space

### ISSUE THREAD

**Issue by rcedgar** [bbuchfink/diamond#357]
Got this demsg with -b 1 on machine with 64Gb RAM:

`Out of memory: Killed process 27429 (diamond) total-vm:69678788kB, anon-rss:64312120kB, file-rss:0kB, shmem-rss:0kB`

I'm screening vertebrate chromosomes against a viral proteome search database so that I can mask out false positive alignments to hosts.  I'm guessing the problem is that diamond can't handle 100Mb+ vertebrate chromosomes as queries so I'll try working around the problem by chopping the chromosomes into overlapping 1kb chunks. If my guess is right, suggest issuing a warning or error from the code because "Killed" is a bit baffling to a user.

**Comment by bbuchfink** [comment]
Long queries can be handled better when using frameshift mode `-F 15` in combination with `--range-culling`, although the maximum I have tried were bacterial chromosomes. Some adjustments may be necessary to handle such big genomes, but your method of chopping it up should also work fine.

Diamond prints a proper error message when running out of memory normally, except when it's killed by the OS. As far as I know, you can't write a handler for SIGKILL so I'm not sure how to address this.

**Comment by rcedgar** [comment]
You could track the query sequence length or amount of memory you're allocating (using /proc on Linux or GetProcessMemoryInfo on Windows) and issue a warning/error if much longer/more than expected. Just a thought, not a problem for me any more.

**Comment by bbuchfink** [comment]
That's true, it can't hurt to have a safeguard in place instead of having users find out the hard way.
